### PR TITLE
[14.0][ADD] product_supplierinfo_stock_picking_type: New addon.

### PR DESCRIPTION
**Pull request by victoralmau**
New addon.

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT38584

### CODE REVIEW

**Comment by pedrobaeza** [comment]
@LudLaf please review README

File: product_supplierinfo_stock_picking_type/models/product_product.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    def _prepare_sellers(self, params=False):
+        res = super()._prepare_sellers(params)
+        if params and params.get("order_id"):
+            return res.filtered(
+                lambda x: x.picking_type_id == params.get("order_id").picking_type_id
```
Comment (pedrobaeza) [inline product_supplierinfo_stock_picking_type/models/product_product.py]:
```suggestion
                lambda x: not x.picking_type_id or x.picking_type_id == params.get("order_id").picking_type_id
```

Comment (pedrobaeza) [reply product_supplierinfo_stock_picking_type/models/product_product.py]: Include a test case for this False one as well.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Review by pedrobaeza — APPROVED** [review]
Code review. README not reviewed

File: product_supplierinfo_stock_picking_type/readme/USAGE.rst [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+Purchase order flow A:
+#. Go to *Purchase > Orders > Requests for Quotation* and create new order as follows:
+#. Vendor: Azure Interior, Deliver To: Incoming A ('Other information' tab), Product: Test product
+#. The unit price of the product will be 5.
+
+Purchase order flow B:
+#. Go to *Purchase > Orders > Requests for Quotation* and create new order as follows:
+#. Vendor: Azure Interior, Deliver To: Incoming B ('Other information' tab), Product: Test product
+#. The unit price of the product will be 10.
```
Comment (LudLaf) [inline product_supplierinfo_stock_picking_type/readme/USAGE.rst]:
```suggestion
#. Go to *Inventory > Configuration > Operations types* and create two records
   named 'Incoming A' and 'Incoming B' with 'Type of Operation' equal to
   'Receipt'.
#. Go to *Inventory > Products > Products*, then create a product named
   'Test product' and set 'Purchase' tab as follows:
#. [Vendor line 1] Vendor: Azure Interior, Picking type: Incoming A, Price: 5.
#. [Vendor line 2] Vendor: Azure Interior, Picking type: Incoming B, Price: 10.
#. Then click on 'Save'.

Next steps:

Purchase order flow A:
#. Go to *Purchase > Orders > Requests for Quotation* and create a new order
   as follows:
#. Vendor: Azure Interior; Product: Test product; Deliver To: Incoming A
   (under 'Other information' tab)
#. The unit price of the product will be 5.

Purchase order flow B:
#. Go to *Purchase > Orders > Requests for Quotation* and create a new order
   as follows:
#. Vendor: Azure Interior; Product: Test product; Deliver To: Incoming B
   (under 'Other information' tab)
#. The unit price of the product will be 10.
```

Comment (LudLaf) [reply product_supplierinfo_stock_picking_type/readme/USAGE.rst]: Replacing this whole text will probably be easier.

**Review by LudLaf — CHANGES_REQUESTED** [review]

**Comment by victoralmau** [comment]
Changes done.

**Review by LudLaf — APPROVED** [review]

**Review by chienandalu — APPROVED** [review]
Tested :+1: (I had to remove all the purchase modules to avoid behavior interferences...)

Naif question: should the prices be updated when we change the picking typein the purchase order?

**Comment by victoralmau** [comment]
> Tested +1 (I had to remove all the purchase modules to avoid behavior interferences...)
> 
> Naif question: should the prices be updated when we change the picking typein the purchase order?

Interesting, good details.

I have added `picking_type_id` field to the form view of `product.supplierinfo`.

I have added the `onchange()` in the purchase order to update the line prices when the `picking_type_id` field is changed.

**Pull request merged**